sonar-db/src/main/java/org/sonar/db/component/ComponentDao.java

```diff
@@ -193,6 +193,15 @@ public ComponentDto selectOrFailByKey(DbSession session, String key) {
     return Optional.fromNullable(mapper(session).selectByKey(key));
   }
 
+  /**
+   * Select all Components in DB with the specified key.
+   *
+   * Used by Developer Cockpit
+   */
+  public List<ComponentDto> selectByKeyWithDuplicates(DbSession session, String key) {
```

Comment:
So you could either use selectComponentsHavingSameKeyOrderedById, or replace usage of selectComponentsHavingSameKeyOrderedById by you're new one but in this case don't forget to add the order by id